packages/@react-aria/table/src/useTableRow.ts

```diff
@@ -42,14 +43,14 @@ export function useTableRow<T>(props: GridRowProps<T>, state: TableState<T> | Tr
   let {rowProps, ...states} = useGridRow<T, TableCollection<T>, TableState<T>>(props, state, ref);
   let {direction} = useLocale();
 
-  if (isVirtualized && !('expandedKeys' in state)) {
+  if (isVirtualized && !(tableNestedRows() && 'expandedKeys' in state)) {
     rowProps['aria-rowindex'] = node.index + 1 + state.collection.headerRows.length; // aria-rowindex is 1 based
   } else {
     delete rowProps['aria-rowindex'];
   }
 
   let treeGridRowProps: HTMLAttributes<HTMLElement> = {};
-  if ('expandedKeys' in state) {
+  if (tableNestedRows() && 'expandedKeys' in state) {
```

Comment:
wanted to just make a var for `let isTreeGrid = tableNestedRows() && 'expandedKeys' in state;` and use it here, but typescript wasn't smart enough to figure out that state would then be a `TreeGridState` only and complained about lines below